service.go

```diff
@@ -86,3 +89,40 @@ func (s *sectionServiceImpl) initSectionsMap(terms []interface{}) {
 		s.sectionLinks = append(s.sectionLinks, sectionLink{APIURL: s.baseURL + top.UUID})
 	}
 }
+
+func (s *sectionServiceImpl) getSectionCount() int {
+	return len(s.sectionLinks)
+}
+
+func (s *sectionServiceImpl) getSectionIds() []string {
+	i := 0
+	keys := make([]string, len(s.sectionsMap))
+
+	for k := range s.sectionsMap {
+		keys[i] = k
+		i++
+	}
+	return keys
+}
+
+func (s *sectionServiceImpl) reload() error {
```

Comment:
This is the duplication of `init()` method, could you please make it less redundant?